httpx/httptest/server.go

```diff
@@ -0,0 +1,151 @@
+package httptest
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/ryanuber/go-glob"
+)
+
+const (
+	// Anything is used where the expectation should not be considered.
+	Anything = "httpx/httptest.Anything"
+)
+
+// Expectation represents an http request expectation.
+type Expectation struct {
+	method string
+	path   string
+
+	fn http.HandlerFunc
+
+	headers []string
+	body    []byte
+	status  int
+
+	times int
+}
+
+// Times sets the number of times the request can be made.
+func (e *Expectation) Times(times int) *Expectation {
+	e.times = times
+
+	return e
+}
+
+// Header sets the HTTP headers that should be returned.
+func (e *Expectation) Header(k, v string) *Expectation {
+	e.headers = append(e.headers, k, v)
+
+	return e
+}
+
+// Handle sets the HTTP handler function to be run on the request.
```

Comment:
They do but `Returns*` and `Handle` are the "end" of the chain. It will always be last in the call, cause you are telling it how to return or handle the call.